File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.java
import java.util.*;

class Solution {
    public int solution(int[] elements) {
        int[] e = new int[elements.length + 1];
        for (int i = 0; i < elements.length; i++) {
            e[i + 1] = e[i] + elements[i];
        }
        
        Set<Integer> answer = new HashSet<>();
        
        for (int left = 0; left < e.length - 1; left++) {
            for (int right = left+1; right < e.length; right++) {
                int total = e[right] - e[left];
                answer.add(total);
                answer.add(e[e.length - 1] - total);
            }
        }
        
        return answer.size() - 1;
    }
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    answer = set(elements)
    
    i = 0 
    
    while i < len(elements):
        current = elements[i]
        answer.add(current)
        j = (i + 1) % len(elements)
        
        while j != i:
            current += elements[j]
            answer.add(current)
            j += 1
            j %= len(elements)
        i += 1
    return len(answer)